src/mdast-process-images.js

```diff
@@ -12,28 +12,78 @@
 import { visit } from 'unist-util-visit';
 import processQueue from '@adobe/helix-shared-process-queue';
 
+function createFilter(log, baseUrlStr, imgSrcPolicy) {
+  const baseUrl = new URL(baseUrlStr);
+  const imgSrcPolicFilters = imgSrcPolicy.map((policyValue) => {
+    // filter according to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/Sources
+    if (policyValue === '*') {
+      return () => true;
+    }
+
+    if (policyValue === 'self') {
+      return (urlStr) => {
+        try {
+          const { host } = new URL(urlStr);
+          return baseUrl.host === host;
+        } catch (e) {
+          log.warn(`Failed to parse url '${urlStr}': ${e.message}`);
+          return false;
+        }
+      };
+    }
+
+    return (urlStr) => {
+      try {
+        // omit the protocol check, as https:// is enforced earlier already
+        // omit the pathname check, as we may not need it (yet)
+        const url = new URL(urlStr);
+        const protocolEndIdx = policyValue.indexOf('://');
+        let host = protocolEndIdx < 0 ? policyValue : policyValue.substring(protocolEndIdx + 3);
+        const slashIdx = host.indexOf('/');
+        if (slashIdx >= 0) host = host.substring(0, slashIdx);
+
+        if (host.startsWith('*')) {
+          // allow subdomain
+          const dotIdx = url.host.indexOf('.');
+          if (dotIdx < 0) return false;
+
+          return url.host.substring(dotIdx) === host.substring(1);
+        } else {
+          return url.host === host;
+        }
+      } catch (e) {
+        log.warn(`Failed to parse url '${urlStr}': ${e.message}`);
+        return false;
+      }
+    };
+  });
+
+  return (urlStr) => urlStr.startsWith('https://') && imgSrcPolicFilters.some((f) => f(urlStr));
+}
+
 /**
  * Process images
  * @param {Console} log
  * @param {object} tree
  * @param {MediaHandler} mediaHandler
  * @param {string} baseUrl
  */
-export async function processImages(log, tree, mediaHandler, baseUrl) {
+export async function processImages(log, tree, mediaHandler, baseUrl, imgSrcPolicy) {
   if (!mediaHandler) {
     return;
   }
 
   // gather all image nodes
+  const filter = createFilter(log, baseUrl, imgSrcPolicy);
   const images = [];
   visit(tree, (node) => {
     if (node.type === 'image') {
       const { url = '' } = node;
-      if (url.startsWith('https://')) {
-        images.push(node);
-      } else if (url.indexOf(':') < 0) {
+      if (url.indexOf(':') < 0) {
```

Comment:
I would always accept all relative urls...

Reply:
Which means we implicitly assume the 'self' img-src policy value and skip the unnecessary filtering in this case. done in https://github.com/adobe/helix-html2md/pull/272/commits/85533c454c1907ee7e7a563b1f31adfadf6e9df6